api/server/router/grpc/grpc.go

```diff
@@ -45,16 +53,17 @@ func (gr *grpcRouter) initRoutes() {
 	}
 }
 
-func unaryInterceptor() grpc.UnaryServerInterceptor {
-	withTrace := otelgrpc.UnaryServerInterceptor() //nolint:staticcheck // TODO(thaJeztah): ignore SA1019 for deprecated options: see https://github.com/moby/moby/issues/47437
+func unaryInterceptor(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp any, err error) {
+	if strings.HasSuffix(info.FullMethod, "opentelemetry.proto.collector.trace.v1.TraceService/Export") {
+		return handler(ctx, req)
+	}
 
-	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
-		// This method is used by the clients to send their traces to buildkit so they can be included
-		// in the daemon trace and stored in the build history record. This method can not be traced because
-		// it would cause an infinite loop.
-		if strings.HasSuffix(info.FullMethod, "opentelemetry.proto.collector.trace.v1.TraceService/Export") {
-			return handler(ctx, req)
+	resp, err = handler(ctx, req)
+	if err != nil {
+		logrus.Errorf("%s returned error: %v", info.FullMethod, err)
+		if logrus.GetLevel() >= logrus.DebugLevel {
```

Comment:
I think we can use the containerd/log package for this as well (we're trying to keep logrus out of our imports);


```suggestion
		if log.GetLevel() >= log.DebugLevel {
```

Wondering though; would regular logging not work here?

```suggestion
		log.G(ctx).Debug()
```

Reply:
Perhaps;

```go
if log.GetLevel() >= log.DebugLevel {
	log.G(ctx).WithError(err).Error(info.FullMethod)
} else {
	log.G(ctx).WithFields(log.Fields{
		"error": err,
		"stack": fmt.Sprintf("%+v", stack.Formatter(grpcerrors.FromGRPC(err))),
	}).Error(info.FullMethod)
}
```

Reply:
Sounds good. I kept the conditional print to `stderr` because I was just concerned about changing the offending deprecated code and didn't want to look into if we could rework it, but changing to containerd/log makes sense, I'll take your (combined) solution.